inc/zoo/swar/associative_iteration.h

```diff
@@ -218,13 +251,57 @@ constexpr auto multiplication_OverflowUnsafe_SpecificBitCount(
 
     auto halver = [](auto counts) {
         auto msbCleared = counts & ~S{S::MostSignificantBit};
-        return S{msbCleared.value() << 1};
+        return S{static_cast<T>(msbCleared.value() << 1)};
     };
 
-    multiplier = S{multiplier.value() << (NB - ActualBits)};
+    multiplier = S{static_cast<T>(multiplier.value() << (NB - ActualBits))};
     return associativeOperatorIterated_regressive(
-        multiplicand, S{0}, multiplier, S{S::MostSignificantBit}, operation,
-        ActualBits, halver
+        multiplicand,
+        S{0},
+        multiplier,
+        S{S::MostSignificantBit},
+        operation,
+        ActualBits,
+        halver
+    );
+}
+
+template<int ActualBits, int NB, typename T>
+constexpr auto expo_OverflowUnsafe_SpecificBitCount(
```

Comment:
I must object to this identifier:
See that I am using unseemly long identifiers looking to make sure the names are clear, there is dissonance with using the `expo` abbreviation for exponentiation while the specification of the flavor of the operation takes 10 times as many characters.

Reply:
Agreed.